contracts/feature-tests/exchange-features/src/exchange_features.rs

```diff
@@ -0,0 +1,77 @@
+#![no_std]
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub const FULL_VALUE_SUPPLY: u64 = 16;
+
+#[derive(
+    ManagedVecItem,
+    TopEncode,
+    TopDecode,
+    NestedEncode,
+    NestedDecode,
+    TypeAbi,
+    Clone,
+    PartialEq,
+    Debug,
+)]
+pub struct TokenAttributes<M: ManagedTypeApi> {
+    pub amount: BigUint<M>,
+}
+
+impl<M: ManagedTypeApi> FixedSupplyToken<M> for TokenAttributes<M> {
+    fn get_total_supply(&self) -> BigUint<M> {
+        self.amount.clone()
+    }
+
+    fn into_part(self, payment_amount: &BigUint<M>) -> Self {
+        let new_amount =
+            self.rule_of_three_non_zero_result(payment_amount, &BigUint::from(FULL_VALUE_SUPPLY));
```

Comment:
Use `self.amount` instead of the const, otherwise, successive `into_part` will give the wrong value.